data_steward/tools/consolidated_achilles_report/get_all_results_html.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import print_function
+import gcs_utils
+import json
+import os
+import query_reports
+
+DRC_BUCKET_PATH = 'gs://%s/' % gcs_utils.get_drc_bucket()
+
+
+def get_hpo_id(p):
+    rel_path = p[len(DRC_BUCKET_PATH):]
+    return rel_path[:rel_path.index('/')]
+
+
+def transform_bq_list(uploads):
+    """
+    Get paths to all most recent report files
+
+    :return:
+    """
+    results = []
+    for upload in uploads:
+        dte, p = upload['upload_timestamp'], upload['file_path']
+        hpo_id = get_hpo_id(p)
+        report_path = p.replace('person.csv', 'results.html')
+        result = dict(hpo_id=hpo_id, updated=dte, report_path=report_path)
+        results.append(result)
+    return results
+
+
+def download_report(s):
+    """
+    Download most recent report files
+    :param s:
+    :return:
+    """
+    # Save it to curation_report/data/<hpo_id>
+    cdir = os.getcwd()
+    if not os.path.exists('%s/result_data' % (cdir)):
```

Comment:
To prevent race conditions and simplify your code, use try/except instead of checking for existence.  I think of this type of try/except as asking forgiveness rather than permission.

```
try:
    os.makedirs('%s/result_data/%s' % (cdir, s['hpo_id']))
except OSError:
    # log the exception but keep moving because it doesn't hurt your code.
    LOGGER.debug("The file %s/result_data/%s already exists", cdir, s['hpo_id'])
```

Then you can remove the if/else statements on lines 42-50 and replace them with `cmd`, `print`, and `os` calls once.
```
cmd = 'gsutil -m cp -r %s ./result_data/%s/' % (s['report_path'], s['hpo_id'])
print('Downloading {} rpt with cmd: `{}`...'.format(s['hpo_id'], cmd))
os.system(cmd)
```

I think there should be a way to do this without issuing a system command, but I'm still looking into it.  Codacy is complaining about the system command lines.

Reply:
yes, this is a great idea, i'll update it.